### PR TITLE
Fixing IT teardown behavior

### PR DESCRIPTION
**Pull request by zen-dog**
Summary:
our teardown was broken on multiple levels (starting with the actual `POST mesos/teardown` call which never worked and ending with our `MesosCluster::teardown` method  which would retry but never wait for the actual teardown to happen.

### CODE REVIEW

File: mesos-simulation/src/test/scala/mesosphere/mesos/scale/SingleAppScalingTest.scala [hunk 1]
```diff
@@ -20,6 +20,14 @@ object SingleAppScalingTest {
   val appInfosFile = "scaleUp20s-appInfos"
 }
 
+trait SimulatedMesosTest extends MesosTest {
```
Comment (jeschkies) [inline mesos-simulation/src/test/scala/mesosphere/mesos/scale/SingleAppScalingTest.scala]: What is the idea of this?

File: src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala [hunk 2]
```diff
@@ -335,37 +335,18 @@ case class MesosCluster(
...
+    val frameworkIds = facade.frameworkIds().value
 
-trait SimulatedMesosTest extends MesosTest {
-  def mesos: MesosFacade = {
-    require(false, "No access to mesos")
-    ???
+    // Call mesos/teardown for all framework Ids in the cluster and wait for the teardown to complete
+    frameworkIds.foreach { fId =>
+      facade.teardown(fId)
+      eventually(timeout(1.minutes), interval(2.seconds)) { facade.completedFrameworkIds().value.contains(fId) }
```
Comment (jeschkies) [inline src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala]: How about moving the assertion out of the loop?
1. Issue treadown for each framework.
2. Verifiy that all frameworks are down.

File: src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala [hunk 3]
```diff
@@ -335,37 +335,18 @@ case class MesosCluster(
...
-              s"agent allocated on agent ${agent.id}: " +
-                s"used = ${agent.usedResources}, reserved = ${agent.reservedResourcesByRole}")
-          }
-        }
-      }
-    Await.result(teardown, Duration.Inf)
-  }
-}
+object MesosTest extends Eventually {
+  def teardown(facade: MesosFacade): Unit = {
```
Comment (jeschkies) [inline src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala]: Do we use this only in `trait Mesos`? Cause then we could move the functionality.

File: mesos-simulation/src/test/scala/mesosphere/mesos/scale/SingleAppScalingTest.scala (hunk 1 above)
Comment (zen-dog) [reply mesos-simulation/src/test/scala/mesosphere/mesos/scale/SingleAppScalingTest.scala]: ¯\_(ツ)_/¯
I merely moved it to the only place where it's used.

File: src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala (hunk 2 above)
Comment (zen-dog) [reply src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala]: 1. happens already
2. not sure what you mean by that. That's what `eventually` is waiting for.

File: src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala (hunk 3 above)
Comment (zen-dog) [reply src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala]: You're right. No reason to keep this object around.

**Comment by zen-dog** [comment]
> No leaked processes detected

First time I see this happen after the test ^.^

**Pull request merged**